### ISSUE TITLE
Memory Leak+ reverts to Memory Leak behaviour

### ISSUE THREAD

**Issue by arganoid** [TerryCavanagh/diceydungeons.com#2316]
See: [https://youtu.be/5AJP-PzAvGc](url)

This was happening consistently on this run. I had the rule which adds 1 Freeze on each turn. Every time I rolled the first dice, the behaviour of Memory Leak+ reverted to Memory Leak, i.e. the +1 was not shown or applied. The title was still shown as Memory Leak+.

v1.10, 05-11-2020, STEAM, 64-bit, GL

**Comment by JuniorUdale** [comment]
I'm pretty sure that's intended - the upgrade means that after 1 calculation, it takes any dice, rather than being max5. So instead of the upgrade giving it +2 damage, in practice it gives it a "buffer" of 1 calculation before its slot starts shrinking.

**Comment by TerryCavanagh** [comment]
Ah, yep, this is intended, as @JuniorUdale says. Actually I think on reflection your interpretation of how it works is clearer and makes more sense, though. What do you think - should I change it?

*Old behaviour:* Calculating causes the slot to reduce from NORMAL to MAX5 to MAX4, etc. When upgraded, the slots start as NORMAL|FREE1 and reduce to NORMAL, MAX5, etc.

*Proposed new behaviour:* Memory Leak+ starts as NORMAL|FREE1, but reduces to MAX5|FREE1, MAX4|FREE1, etc.

**Comment by arganoid** [comment]
Yes, I think the proposed new behaviour better matches the equipment description.
